content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+This effectively emulates a mechanism to _"re-activate"_ a stale Noise session by binding it to a newly created active Noise session.
+
+In order to reduce users' metadata leakage, it is desirable (as suggested in [35/WAKU2-NOISE](https://rfc.vac.dev/spec/35/#after-handshake)) that content topics used for communications change every time a new message is exchanged.
+This can be easily realized by employing a key derivation function to compute a new `session-id` from the previously employed one (e.g. `session-id = HKDF(prev-session-id)`),
+while keeping the Inbound/outbound Cipher States, the content topic derivation mechanism and the stale mechanism the same as above.
+In this case, when one party sends **and** receives at least one message, 
+he SHALL publicly mark as stale all Noise sessions relative to messages exchanged before the earlier of these two send/receive events.
+
+# Multi-Device support
+
+Alice and Bob might possess one or more devices (e.g. laptops, smartphones, etc.) they wish to use to communicate.
+In the following we assume Alice and Bob to possess $N$ and $M$ devices, respectively.
+
+Since a Noise session contains cryptographic material required to encrypt and decrypt messages exchanged on a pre-defined content topic derived from a `session-id`, 
+messages should be encrypted and decrypted within the Noise session instantiated between the currently-in-use sender's and receiver's device.
+
+This is achieved through two main supported session management mechanisms that we called `N11M` and `NM`, respectively.
+
+## The $N11M$ session management mechanism
+
+In a $N11M$ setting, each party's device shares the same Noise session information used to encrypt and decrypt messages exchanged with the other party.
+
+![](./N11M.png)
+
+More precisely, once the first Noise session between any of Alice's and Bob's device is instantiated, 
+its session information is securely propagated to all other devices, 
+which then become able to send and receive new messages on the content topic associated to such session.
+
+The most updated session information between Alice and Bob is propagated in encrypted form to other devices, 
+using previously instantiated Noise sessions. 
+The propagated session information corresponds to the latest session information stored on the device currently communicating with (any of the devices of) the other party.
+
+We note that sessions information is propagated only among devices belonging to the same party and not with other party's devices. 
+Hence, Alice has no knowledge on the number of devices Bob is using and vice versa. 
+
+When any device marks a Noise session between Alice and Bob as stale, 
+all (updated) devices will implicitly mark such session as stale.
+
+In case a Noise session between two devices belonging to the same party is marked as stale, 
+such two devices stop to reciprocally propagate any information regarding Noise sessions instantiated with other parties.
```

Comment:
What is the assumption about the handshake phase between the two parties' devices? let's say 2 devices of Alice simultaneously enter a handshake with one or more devices of Bob? they all arrive at the same content topic and they will collide.

By the way, it would be interesting and useful to look into scenarios where devices of the same person can be online at the same time and not in sync with each other.  How devices of the same person are differentiated?

Reply:
> What is the assumption about the handshake phase between the two parties' devices? let's say 2 devices of Alice simultaneously enter a handshake with one or more devices of Bob? they all arrive at the same content topic and they will collide.

Collisions are practically impossible as long as the handshake involves DH operations over ephemeral keys (not considering that each device may have a different static key too). If two devices simultaneously instantiate a session with the other party, these two party-to-party sessions will have two distinct session-id and Cipher states pairs, in turn shared among all parties' devices. Having two concurrent session with one party is a requirement for some applications, e.g. being able to download multiple files from the same source.

> By the way, it would be interesting and useful to look into scenarios where devices of the same person can be online at the same time and not in sync with each other. How devices of the same person are differentiated?

There is no need to. Simultaneous messages from different devices will be automatically ordered thanks to increasing Cipher States nonces, that is if are encrypted with same key and same nonce only the first one received will be correctly decrypted (you may implement an ACK mechanism on top, but this is left to the application layer). See https://github.com/vacp2p/rfc/pull/504#discussion_r878679430 for a related comment. If a device session information is not synced with the other party's session, received messages will simply be discarded as long as this device does not get the most updated session information shared by other devices.

Reply:
> Collisions 

Here is what I meant by collision:
Alice has two devices A1, and A2, where A1 and A2 are not synchronized
Bob has one device B1
A1 does the handshake with B1 and enters a Noise session with B1
A2 also does a handshake with B1 and enters a Noise session  with B1
Now, A1 and A2 are concurrently talking with B1 in two separate sessions
From B1's point of view, it is talking with Alice on two different Noise sessions. Now, assume this is a chat app, it is like Bob is talking to two Alices. It might make sense for Bob i.e., B1 to check whether it already has a connection with Alice or not and e.g., inform A2 of such communication OR refuses to instantiate a connection with A2 if there is another active session. Although, this can be an application concern. Please let me know if that makes sense. 

BTW,  I meant to consider such scenarios in the future (not in this PR).